src/industry_gui.cpp

```diff
@@ -1430,12 +1440,26 @@ class IndustryDirectoryWindow : public Window {
 	 */
 	static int GetCargoTransportedSortValue(const Industry *i)
 	{
-		int p1 = GetCargoTransportedPercentsIfValid(i, 0);
-		int p2 = GetCargoTransportedPercentsIfValid(i, 1);
-
-		if (p1 > p2) Swap(p1, p2); // lower value has higher priority
+		CargoID filter = IndustryDirectoryWindow::produced_cargo_filter;
+
+		int p = 0, c = 0;
```

Comment:
I would like to see some comments about what is going on here in that function. And what `p`, `c`, & `t` is for? :)